tests/CMakeLists.txt

```diff
@@ -1,34 +1,33 @@
-INCLUDE_DIRECTORIES("${CMAKE_CURRENT_SOURCE_DIR}")
-INCLUDE_DIRECTORIES("${CMAKE_CURRENT_BINARY_DIR}")
-INCLUDE_DIRECTORIES("${CMAKE_SOURCE_DIR}/include")
-INCLUDE_DIRECTORIES("${CMAKE_BINARY_DIR}")
-INCLUDE_DIRECTORIES("${CMAKE_BINARY_DIR}/src")
+include(CTest)
 
-SET(CMAKE_CXX_STANDARD 17)
-
-SET(CMAKE_AUTOMOC ON)
-
-# FIXME: remove this once we export include directories for LMMS
-IF(LMMS_BUILD_APPLE)
-INCLUDE_DIRECTORIES("/usr/local/include")
-ENDIF()
-
-ADD_EXECUTABLE(tests
-	EXCLUDE_FROM_ALL
-	main.cpp
-	QTestSuite.cpp
-	$<TARGET_OBJECTS:lmmsobjs>
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+set(CMAKE_AUTOMOC ON)
 
+set(LMMS_TESTS
```

Comment:
A wildcard search would be cool, because otherwise, some people might add their tests, just run ctest and read "all tests passed, that must include my test". Though I think that is unlikely :smiling_face_with_tear:  On the other hand, for LMMS source code, we do not use wildcards either, so this might be good consistency.

Reply:
Wildcard search would imply globbing, which CMake goes against in the docs:
> We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.

So it actually might me more of a headache to do globbing here, because when someone creates a new test, CMake might not be able to tell the build system "hey, a new test was added", which would then actually prevent the test from being added.

Reply:
Interesting point :+1: